src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/InitializeHandler.cs

```diff
@@ -313,6 +319,12 @@ private bool GetMergedRenameProvider()
             return _serverCapabilities.Any(s => s.Capabilities.RenameProvider?.Value is bool isRenameSupported && isRenameSupported);
         }
 
+        private VSInternalInlineCompletionOptions GetMergedInlineCompletionProvider()
+        {
+            var regexes = _serverCapabilities.Where(s => s.Capabilities.InlineCompletionOptions != null).Select(s => s.Capabilities.InlineCompletionOptions.Pattern.ToString());
```

Comment:
I'm not super up to date on this class, do we expect (now or in the future) there to be more than one `_serverCapability` that matches this? If not we should change this to `.First()`. Also what happens if `InlineCompletionOptions.Pattern` is null?

Reply:
yeah there definitely could be - for example C# and html both providing snippets and we would want to trigger on keywords from both.  If the pattern is null it just means it provides no keywords.

Reply:
> If the pattern is null it just means it provides no keywords.

I more meant that since `Pattern` might be null or un-assigned that calling `Pattern.ToString()` might cause an exception.